### PR TITLE
Finish deprecating idle_event; style cleanups to backend_bases

### PR DESCRIPTION
**Pull request by anntzer**
All's in the title.

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -1363,12 +1354,12 @@ def __init__(self, name, canvas, guiEvent=None):
         self.guiEvent = guiEvent
 
 
+@cbook.deprecated("2.1")
```
Comment (dstansby) [inline lib/matplotlib/backend_bases.py]: Can this deprecation warning have some more information? (eg. along the lines of 'idle_event is deprecated. Use the animations module instead.')

Would it also be worth putting in a what's new entry, saying this is deprecated and pointing people to what to use instead?

File: lib/matplotlib/backend_bases.py [hunk 2]
```diff
@@ -1982,6 +1953,7 @@ def enter_notify_event(self, guiEvent=None, xy=None):
         event = Event('figure_enter_event', self, guiEvent)
         self.callbacks.process('figure_enter_event', event)
 
+    @cbook.deprecated("2.1")
```
Comment (dstansby) [inline lib/matplotlib/backend_bases.py]: Ditto my above comments

**Comment by anntzer** [comment]
Actually the main entry point (connecting to "idle_event") already has a deprecation warning with a pointer to the animations module since 1.5 (http://matplotlib.org/api/api_changes.html?highlight=idle_event#deprecated-idle-event, https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backend_bases.py#L2352) so I don't think another warning is needed here.

**Comment by dstansby** [comment]
One final question; should this be targeted to `2.0.x`? (I don't think it should) - if not, I think there should probably be at least one cycle of deprecation, so the deprecation warning should be `2.2`

**Comment by anntzer** [comment]
2.1 is not released yet so it's OK to say "deprecated since 2.1".  It's not a removal date.

**Comment by dstansby** [comment]
🤦‍♂️ I keep forgetting that, sorry...

**Pull request merged**